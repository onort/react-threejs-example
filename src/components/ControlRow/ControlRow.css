.row {
  margin: var(--spacer-sm) 0;
}

.inline {
  display: flex;
  align-items: center;
}

.label {
  font-size: var(--text-lg);
}

.inline .label {
  margin-right: var(--spacer-md);
}

.container {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: var(--spacer-xsm) 0;
}
