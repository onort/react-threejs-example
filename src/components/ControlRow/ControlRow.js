import React from "react"
import PropTypes from "prop-types"

import styles from "./ControlRow.css"

const ControlRow = props => {
  return (
    <div className={`${styles.row} ${props.inline ? styles.inline : ""}`}>
      <span className={styles.label}>{props.label}</span>
      <div className={styles.container}>{props.children}</div>
    </div>
  )
}

ControlRow.proptypes = {
  inline: PropTypes.bool,
  label: PropTypes.string.isRequired
}

ControlRow.defaultProps = {
  inline: false
}

export default ControlRow
