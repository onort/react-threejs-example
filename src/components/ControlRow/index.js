import ControlRow from "./ControlRow"

export default ControlRow
