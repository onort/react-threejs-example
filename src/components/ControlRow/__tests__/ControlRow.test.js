import React from "react"
import { shallow } from "enzyme"

import ControlRow from "../"

const testLabel = "testLabel"

describe("<ControlRow />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<ControlRow label={testLabel} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render label", () => {
    const wrapper = shallow(<ControlRow label={testLabel} />)
    expect(wrapper.find(".label").exists()).toBe(true)
    expect(wrapper.find(".label").text()).toBe(testLabel)
  })
})
