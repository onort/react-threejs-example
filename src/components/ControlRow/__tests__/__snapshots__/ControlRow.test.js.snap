// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlRow /> should match snapshot 1`] = `
<div
  className="row"
>
  <span
    className="label"
  >
    testLabel
  </span>
  <div
    className="container"
  />
</div>
`;

exports[`<ControlRow /> should match snapshot 2`] = `"<div class=\\"row\\"><span class=\\"label\\">testLabel</span><div class=\\"container\\"></div></div>"`;
